cmd/image/imageImport.go

```diff
@@ -86,24 +98,37 @@ So if a file './rancher/k3d-tools' exists, k3d will try to import it instead of
 
 	cmd.Flags().BoolVarP(&loadImageOpts.KeepTar, "keep-tarball", "k", false, "Do not delete the tarball containing the saved images from the shared volume")
 	cmd.Flags().BoolVarP(&loadImageOpts.KeepToolsNode, "keep-tools", "t", false, "Do not delete the tools node after import")
-
+	cmd.Flags().StringP("mode", "m", string(k3d.ImportModeAutoDetect), "Which method to use to import images into the cluster [auto, direct, tools]. See https://k3d.io/usage/guides/importing_images/")
 	/* Subcommands */
 
 	// done
 	return cmd
 }
 
 // parseLoadImageCmd parses the command input into variables required to create a cluster
-func parseLoadImageCmd(cmd *cobra.Command, args []string) ([]string, []k3d.Cluster) {
+func parseLoadImageCmd(cmd *cobra.Command, args []string) ([]string, []*k3d.Cluster) {
 
 	// --cluster
 	clusterNames, err := cmd.Flags().GetStringArray("cluster")
 	if err != nil {
 		l.Log().Fatalln(err)
 	}
-	clusters := []k3d.Cluster{}
+	clusters := []*k3d.Cluster{}
 	for _, clusterName := range clusterNames {
-		clusters = append(clusters, k3d.Cluster{Name: clusterName})
+		clusters = append(clusters, &k3d.Cluster{Name: clusterName})
+	}
+
+	// Figure out the nodes for each cluster
+	nodeList, err := client.NodeList(cmd.Context(), runtimes.SelectedRuntime)
+	if err != nil {
+		l.Log().Fatalf("Failed to list clusters %v", err)
+	}
+	for _, node := range nodeList {
+		for _, cluster := range clusters {
+			if cluster.Name == node.RuntimeLabels[k3d.LabelClusterName] && node.Role == k3d.ServerRole {
+				cluster.Nodes = append(cluster.Nodes, node)
+			}
+		}
```

Comment:
```suggestion
		cluster, err := client.ClusterGet(cmd.Context(), runtimes.SelectedRuntime, &k3d.Cluster{Name: clusterName})
		if err != nil {
			l.Log().Fatalf("failed to get cluster %s: %v", clusterName, err)
		}
		clusters = append(clusters, cluster)
```

Reply:
@sbaier1 , `ClusterGet` will already populate `cluster.Nodes` and even more information (network, etc.).
Before this was done in `EnsureToolsNode` in a later step, which first gathers some information about the cluster, if not present :+1: